### PR TITLE
TcpCodecs factory improvement

### PR DESCRIPTION
**Pull request by garyrussell**
- add convenience factory methods to set the max message size while creating the codec

### CODE REVIEW

File: spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/serializer/AbstractByteArraySerializer.java [hunk 1]
```diff
@@ -39,7 +39,13 @@
...
 		ApplicationEventPublisherAware {
 
-	protected int maxMessageSize = 2048;
+	/**
+	 * The default maximum message size when deserializing.
+	 * @since 5.1.3
+	 */
+	public static final int DEFAULT_MAX_MESSAGE_SIZE = 2048;
+
+	protected int maxMessageSize = DEFAULT_MAX_MESSAGE_SIZE;
```
Comment (artembilan) [inline spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/serializer/AbstractByteArraySerializer.java]: Isn't Sonar going to object that we continue to keep it as a `protected`?
Although I guess the `private` and `protected` getter is going to be as a breaking change in the point release...

**Review by artembilan — CHANGES_REQUESTED** [review]
Only one comment which may deserve a separate issue for the next version.

File: spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/serializer/AbstractByteArraySerializer.java (hunk 1 above)
Comment (garyrussell) [reply spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/serializer/AbstractByteArraySerializer.java]: Doh - I meant it to be public - this explains why I couldn't get javadocs `@value` to work.

Fixing...

Comment (garyrussell) [reply spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/serializer/AbstractByteArraySerializer.java]: OK; will fix.

**Pull request merged**